python3/httplib2/__init__.py

```diff
@@ -915,28 +930,16 @@ def __init__(self, host, port=None, key_file=None, cert_file=None,
                  timeout=None, proxy_info=None,
                  ca_certs=None, disable_ssl_certificate_validation=False):
 
-        if not hasattr(ssl, 'SSLContext'):
-            raise RuntimeError("httplib2 requires Python 3.2+ for ssl.SSLContext")
-
         self.disable_ssl_certificate_validation = disable_ssl_certificate_validation
-        self.ssl_version = ssl.PROTOCOL_TLS
+        self.ca_certs = ca_certs if ca_certs else CA_CERTS
 
         if proxy_info:
             if isinstance(proxy_info, ProxyInfo):
                 self.proxy_info = proxy_info
             else:
                 self.proxy_info = proxy_info('https')
 
-        context = ssl.SSLContext(self.ssl_version)
-        context.verify_mode = ssl.CERT_NONE if disable_ssl_certificate_validation else ssl.CERT_REQUIRED
-        context.check_hostname = not disable_ssl_certificate_validation
-
-        self.ca_certs = ca_certs if ca_certs else CA_CERTS
-        context.load_verify_locations(self.ca_certs)
-
-        if cert_file:
-            context.load_cert_chain(cert_file, key_file)
-
+        context = _build_ssl_context(self.disable_ssl_certificate_validation, ca_certs, cert_file, key_file)
```

Comment:
This makes it by default use system default certificates instead of CA_CERTS. Fix:

`context = _build_ssl_context(self.disable_ssl_certificate_validation, `**self.**`ca_certs, cert_file, key_file)`

I think it would be better to drop `self.ca_certs` altogether and just rewrite it like `ca_certs = ca_certs or CA_CERTS` but it breaks public API, so not today.

Reply:
`_build_ssl_context` **also** falls back to `CA_CERTS` if the ca_certs param is None. So changing it to `self.ca_certs` would not change anything. See:

https://github.com/httplib2/httplib2/pull/90/files#diff-72a22fe2efc9771bbe023ba1ffa07173R151